packages/infra/lib/shared/metriport-composite-stack.ts

```diff
@@ -0,0 +1,23 @@
+import * as cdk from "aws-cdk-lib";
+import { CfnElement } from "aws-cdk-lib";
+import { Construct } from "constructs";
+
+export class MetriportCompositeStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+  }
+
+  /**
+   * Override to ensure nested stack resources are readable in AWS Console.
+   */
+  public override getLogicalId(element: CfnElement): string {
+    if (element.node.id.includes("NestedStackResource")) {
+      const match = /([a-zA-Z0-9_-]+)\.NestedStackResource/.exec(element.node.id);
+      if (match && match[1]) {
+        return match[1];
+      }
+    }
+
+    return super.getLogicalId(element);
+  }
```

Comment:
What's the situation this prevents/problem it solves?

Reply:
This reminds me that it'd be good to have the nested stacks named as such - it helps if you're looking for a stack vs a nested stack.

Reply:
This fixes the super noisy stack names that are running around our AWS account.

```
APIInfrastructureStack-TerminologyServerNestedStackNestedStackTerminologyServerNestedS-7V26PFSHXDEA
```

becomes

```
APIInfrastructureStack-TerminologyServerNestedStack-7V26PFSHXDEA
```

As far as the nested naming, e.g. Instead of NetworkStack we'd call it NetworkNestedStack?